doc/neps/nep-0028-website-redesign.rst

```diff
@@ -0,0 +1,302 @@
+welcoming.
+
+On the other hand, there are good reasons why few projects have a
+multi-lingual site. It's potentially a lot of extra work. Extra work for
+maintainers is costly - they're already struggling to keep up with the work
+load. Therefore we have to very carefully consider whether a multi-lingual
+site is feasible and weight costs and benefits.
+
+We start with an assertion: maintaining translations of all documentation, or
+even the whole user guide, as part of the NumPy project is not feasible. One
+simply has to look at the volume of our documentation and the frequency with
+which we change it to realize that that's the case. Perhaps it will be
+feasible though to translate just the top-level pages of the website. Those
+do not change very often, and it will be a limited amount of content (order
+of magnitude 5-10 pages of text).
+
+We propose the following requirements for adding a language:
+
+- The language must have a dedicated maintainer
+- There must be a way to validate content changes (e.g. a second
+  maintainer/reviewer, or high quality language support in a freely
+  available machine translation tool)
+- The language must have a reasonable size target audience (to be
+  assessed by the NumPy maintainers)
+
+Benefits include:
+
+- Better serve many existing and potential users
+- Potentially attract a culturally and geographically more diverse set of contributors
+
+Tradeoffs:
+
+- Cost of maintaining a more complex code base
+- Cost of making decisions about whether or not to add a new language
+- Higher cost to making content changes, creates work for language maintainers
+- Any content change should be rolled out with enough delay to have translations in place
+
+Can we define a small enough set of pages and content that it makes sense to do this?
+Probably yes.
+
+Is there an easy to use tool to maintain translations and add them to the website?
+To be discussed - it needs investigating, and may depend on the choice of static site
+generator. One potential option is Crowdin_, which is free for open source projects.
+
+
+Backward compatibility
+----------------------
+
+Given a static site generator is chosen, we will migrate away from Sphinx for
+numpy.org (the website, *not including the docs*). The current deployment can
+be preserved until a future deprecation date is decided (potentially based on
+the comfort level of our new site).
+
+All site generators listed above have visibility into the HTML and Javascript
+that is generated, and can continue to be maintained in the event a given
+project ceases to be maintained.
+
+
+Alternatives
+------------
+
+Alternatives we considered:
+
+1. *Update current site.* A new Sphinx theme could be chosen. This would likely
+   take the least amount of resources initially, however, Sphinx does not have
+   the features we are looking for moving forward such as i18n, responsive design,
+   and a clean, modern look.
```

Comment:
Possibly not the NEP to contain this remark, but - updating the sphinx theme is an option we could consider orthogonally for the rest of the docs

Reply:
Good point. @Shekharrajak has been looking into this. The current theme is starting to feel a little dated.

Reply:
I believe, Static Site for the landing page (and all other static content) can be written in Markdown (.md) file and the  Python API documents (which is generated using Sphinx documentation processing tool) should be generated using Sphinx(because Sphinx is best for Python API documentation generator) only. 

We will be having new (maybe similar to the landing page theme) styling.

The motive behind Markdown is :

1. Easy to edit the static contents. So more users/contributors can be engaged if they want something to add/modify.
2. NumPy team can add blogs or news easily and it will be deployed immediately using Github Hook. (Sphinx generated pages will remain the same. It will be changed only when there is new release).
3. Most of the Static Site Generators are using Markdown (Here Jekyll, Hugo, docusaurus), so if in future we want to move to other frameworks then we just need to move the .md files accordingly.

Reply:
> updating the sphinx theme is an option we could consider orthogonally for the rest of the docs

added this as a note